harness/benchmarks/scripts/Main.gd

```diff
@@ -1,59 +1,108 @@
 ]
 
 func _init():
-    var args = __parse_args()
-    print("Parsed arguments: %s" % str(args))
-    var report = Report.new(args.commit)
-    var factory = Benchmarks.new()
-    for cls in classes:
-        for lang in languages:
-            for benchmark in factory.create(lang, cls):
-                __run_benchmark(benchmark, report)
-    __save_report(report)
-
-
-func __run_benchmark(benchmark, report, iterations=3, runs=10, warmups=15):
-    var stats = Stats.new()
-    print("Running benchmark: %s" % str(benchmark))
-    for warmup in range(warmups):
-        __do_run(warmup, benchmark, stats, runs, true)
-    for iteration in range(iterations):
-        __do_run(iteration, benchmark, stats, runs, false)
-    print("Results: %s" % str(stats))
-    report.add(benchmark, stats.get_results())
-
-
-func __do_run(iteration, benchmark, stats, runs, is_warmup):
-    for run in range(runs):
-        var start = OS.get_ticks_usec()
-        benchmark.exec()
-        var duration = OS.get_ticks_usec() - start
-        if not is_warmup:
-            stats.add(duration)
-            #print("[iteration=%d,run=%d] %dus" % [iteration, run, duration])
-
-func __save_report(report):
-    var f = File.new()
-    f.open("res://build/benchmark-results.json", File.WRITE_READ)
-    f.store_string(report.to_json())
-    f.close()
-
-func __parse_args():
-    var arguments = {}
-    for argument in OS.get_cmdline_args():
-        if argument.find("=") > -1:
-            var key_value = argument.split("=")
-            arguments[key_value[0].lstrip("--")] = key_value[1]
-    return arguments
\ No newline at end of file
+	var args: Dictionary = __parse_args()
+	print("Parsed arguments: %s" % str(args))
+
+	if args.has("iterations"):
+		iterations = args.iterations
+	if args.has("warmups"):
+		warmups = args.warmup
+	if args.has("commit"):
+		commit = args.commit
+	else:
+		commit = str(OS.get_datetime().hash())
+			
+	var report = Report.new(commit)
+	
+	for cls in classes:
+		print("Starting class test: %s" % cls)
+		var benchmarks: Array = Benchmarks.create(languages, cls)
+		var test_name: String = ""
+		for benchmark in benchmarks:
+			var new_test_name = benchmark.name
+			if test_name != new_test_name:
+				test_name = new_test_name
+				print("")
+				print("Running benchmark: %s" % test_name)
+			__run_benchmark(benchmark, report)
+	print("Benchmark tests are over!")
+	__save_report(report)
+
+
+func __run_benchmark(benchmark: Benchmark, report: Report):
+	var stats = Stats.new()
+	if benchmark.warmup:
+		for warmup in range(warmups):
+			__do_run(warmup, benchmark, stats, true)
+	for iteration in range(iterations):
+		__do_run(iteration, benchmark, stats, false)
+	printt("%s:" % benchmark.lang.name, str(stats))
+	report.add(benchmark, stats.get_results())
+
+
+func __do_run(iteration: int, benchmark: Benchmark, stats: Stats, is_warmup: bool):
+	var start: float = OS.get_ticks_usec()
+	benchmark.exec()
```

Comment:
why not use iteration here as a loop?

Reply:
Because gdscript loops cost a lot as we can see in other bench. Si to avoid to mesure the cost of loop and so be more precise, we manually wrote calls.

Reply:
Understandable. But maybe add a comment explaining this?

Reply:
I agree with this, @CedNaru can you fix benchmark, it was more your modifications IIRC.

Reply:
Done